src/main/io/osd.h

```diff
@@ -181,7 +186,9 @@ STATIC_ASSERT(OSD_WARNING_COUNT <= 16, osdwarnings_overflow);
 
 typedef struct osdConfig_s {
     uint16_t item_pos[OSD_ITEM_COUNT];
-
+#ifdef USE_OSD_PROFILES
```

Comment:
@mikeller For "Why does the OSD profile entry have to go at the end?"  
It doesn't have to be at the end, it just happens to be at the end because it is added to the end of adjustmentFunction_e.   It's position in the array must align with its position in adjustmentFunction_e.  I didn't want to change the order of the adjustmentFunction_e enum because I don't know what assumptions the configurator makes about the order and also if someone uses an earlier version of the configurator, they may not get the configuration they want because the order of the enum in that version of the configurator is different.

Without the missing entries in the defaultAdjustmentConfigs array the alignment is broken and  lines like the following is incorrect::
const adjustmentConfig_t *adjustmentConfig = &defaultAdjustmentConfigs[adjustmentRange->adjustmentFunction - ADJUSTMENT_FUNCTION_CONFIG_INDEX_OFFSET];

I'll add the comment to make this clear.

Reply:
Looking at #6608, I think the problem you are solving by filling in the missing entries can also be solved by changing the code in question to be as follows:

```
        const uint8_t channelIndex = NON_AUX_CHANNEL_COUNT + adjustmentRange->auxSwitchChannelIndex;
        if ((rcData[channelIndex] != lastRcData[index]) &&
            adjustmentRange->adjustmentCenter &&
            isRangeActive(adjustmentRange->auxChannelIndex, &adjustmentRange->range)) {
            const adjustmentConfig_t *adjustmentConfig = &defaultAdjustmentConfigs[adjustmentRange->adjustmentFunction - ADJUSTMENT_FUNCTION_CONFIG_INDEX_OFFSET];

            // If setting is defined for step adjustment and center value has been specified, apply values directly (scaled) from aux channel
            if (adjustmentConfig->mode == ADJUSTMENT_MODE_STEP) {
                int value = (((rcData[channelIndex] - PWM_RANGE_MIDDLE) * adjustmentRange->adjustmentScale) / (PWM_RANGE_MIDDLE - PWM_RANGE_MIN)) + adjustmentRange->adjustmentCenter;
```

Reply:
@mikeller Hmmm, I don't think so.  The problem is that the index of the entries in the defaultAdjustmentConfigs array must correspond to the entries in the adjustmentFunction_e enum.  The other option is to find the index by searching the array for a matching adjustmentFunction_e  value, but that doesn't see very efficient.  The other option is to move ADJUSTMENT_OSD_PROFILE to just above the missing entries in the array and the enum, but  that will break the Configurator.  Seems in this case it is best to add the missing entries and be done with it.